src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,274 @@
+            }
+
+            // hide toast
+            if (ToastOptions.Duration > 0)
+            {
+                await HideToastIn(ToastOptions.Duration);
+            }
+        }
+
+        /// <summary>
+        ///     Closes the displaying Toast control
+        /// </summary>
+        /// <returns>A void <c>Task</c></returns>
+        public async Task HideToast()
+        {
+            // get current page
+            var currentPage = GetCurrentContentPage();
+            if (currentPage == null)
+            {
+                return;
+            }
+
+            await HideToast(currentPage);
+        }
+
+        private async Task HideToast(ContentPage currentPage)
+        {
+            CancellationSource.Cancel();
+
+            // get toast view, can be only one or none
+            var toastContainer = FindByName(currentPage.Id.ToString());
+            var toastView = toastContainer?.Children.FirstOrDefault(w => w.GetType() == typeof(ToastView));
+            if (toastView == null)
+            {
+                return;
+            }
+
+            // animate toast
+            if (ToastOptions.CloseAnimation != null)
+            {
+                await ToastOptions.CloseAnimation((ToastView)toastView);
+            }
+
+            // remove toast
+            toastContainer.Children.Remove(toastView);
+        }
+
+        private async Task<Grid?> GetToastContainerSettingUpIfNeededAsync()
+        {
+            // get current page
+            var currentPage = GetCurrentContentPage();
+            if (currentPage == null)
+            {
+                return null;
+            }
+
+            // try get toast container
+            var toastContainer = FindByName(currentPage.Id.ToString());
+            if (toastContainer != null) // found toast container
+            {
+                // check opened toasts, can be only one or none
+                var oldToast = toastContainer.Children.FirstOrDefault(w => w.GetType() == typeof(ToastView));
+                if (oldToast != null) // close old toast
+                {
+                    CancellationSource.Cancel();
+                    toastContainer.Children.Remove(oldToast);
+                }
+            }
+            else // no toast container
+            {
+                // create and register toast container
+                toastContainer = new Grid();
+                RegisterName(currentPage.Id.ToString(), toastContainer);
+
+                // old content
+                var oldContent = currentPage.Content;
+
+                // set new content
+                await MainThread.InvokeOnMainThreadAsync(() => { currentPage.Content = toastContainer; });
+                toastContainer.Children.Add(oldContent);
+            }
+
+            return toastContainer;
+        }
+
+        private static ContentPage? GetCurrentContentPage()
+        {
+            if (Application.Current.MainPage is ContentPage contentPage)
```

Comment:
Create a field for Application.Current.MainPage at the top of the method in case the consumer swaps MainPage

Reply:
@Illedan, the toast should use the active `Application.Current.MainPage`, so would it be necessary?

Reply:
@Illedan, the toast should use the active `Application.Current.MainPage`, so would it be necessary?